File: src/main/java/shopping/wish/infrastructure/persistence/jpa/WishRepositoryImpl.java
package shopping.wish.infrastructure.persistence.jpa;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;
import shopping.wish.domain.WishEntity;
import shopping.wish.domain.WishRepository;

import java.util.List;
import java.util.Optional;

@Repository
@RequiredArgsConstructor
public class WishRepositoryImpl implements WishRepository {

    private final WishJpaRepository wishJpaRepository;

    @Override
    public WishEntity save(WishEntity wish) {
        return wishJpaRepository.save(wish);
    }

    @Override
    public Optional<WishEntity> findByIdAndMemberId(Long wishId, Long memberId) {
        return wishJpaRepository.findByIdAndMemberId(wishId, memberId);
    }

    @Override
    public List<WishEntity> findAllByMemberId(Long memberId) {
        return wishJpaRepository.findAllByMemberId(memberId);
    }

    @Override
    public boolean existsByMemberIdAndProductIdWithLock(Long memberId, Long productId) {
        return wishJpaRepository.findByMemberIdAndProductId(memberId, productId).isPresent();
    }

    @Override
    public void delete(WishEntity wish) {
        wishJpaRepository.delete(wish);
    }
}

File: src/main/kotlin/shopping/infrastructure/wish/WishRepositoryImpl.kt
package shopping.infrastructure.wish

import org.springframework.stereotype.Component
import shopping.domain.wish.Wish
import shopping.domain.wish.WishRepository

@Component
class WishRepositoryImpl(
    private val wishJpaRepository: WishJpaRepository,
) : WishRepository {
    override fun findByMemberId(memberId: Long): List<Wish> = wishJpaRepository.findByMemberId(memberId)

    override fun findByMemberIdAndProductId(
        memberId: Long,
        productId: Long,
    ): Wish? = wishJpaRepository.findByMemberIdAndProductId(memberId, productId)

    override fun existsByMemberIdAndProductId(
        memberId: Long,
        productId: Long,
    ): Boolean = wishJpaRepository.existsByMemberIdAndProductId(memberId, productId)

    override fun save(wish: Wish): Wish = wishJpaRepository.save(wish)

    override fun deleteById(id: Long) {
        wishJpaRepository.deleteById(id)
    }
}
